### DISCUSSION TITLE
is Copilot is best option for beginners

### DISCUSSION THREAD

**Discussion by muhammadullah1** [community/community#11064] [Copilot Conversations]
is Copilot is best option for beginners

**Comment by ahmettulutas** [comment]
My personal short answer is no. As a programming student I personally believe in muscle memory. You can not write codes independently if you only watch coding videos. Similarly, if you are a complete beginner, I wouldn't recommend you to use copilot all the time since copilot is very strong when predicting simple code blocks. However, you main purpose should be designing and working on your muscle memory to write code blocks. Sorry Github. Copilot is awesome but newbies need practise on programmatic thinking and muscle memory.

**Reply by quancntt05** [reply]
I agree with your thinking. Copilot isn't for a beginner

**Comment by MrNocole** [comment]
I think the same as _ahmettulutas_ ,copilot is just a tool to make you code easier,but it may make mistakes,sometimes it is not so easy to found if you dont make enongh practice.

**Comment by konradre** [comment]
I think you can learn faster with it if you use it the right way. Not just letting it auto complete all your code, but using it intelligently to find out why a solution is the best one. It can save a lot of time hunting for the answer to a coding problem online, especially if you cant find an answer to a very specific block of problematic code.

**Comment by levyidan1** [comment]
In no way should you use Copilot before you have the relevant experience needed to write anything that Copilot can write for you. Not only will your brain not know how to code from scratch, but you probably won't have any idea what you're doing and it is very possible that Copilot will generate code with critical errors that you won't notice unless you have the relevant experience.
